testsuite/MDAnalysisTests/core/test_atomgroup.py

```diff
@@ -120,6 +121,76 @@ def test_write_coordinates(self, u, tmpdir):
         with tmpdir.as_cwd():
             u.atoms.write("test.xtc")
 
+    @pytest.mark.parametrize('frames', (
+        [4],
+        [2, 3, 3, 1],
+        slice(2, 6, 1),
+    ))
+    def test_write_frames(self, u, tmpdir, frames):
+        destination = str(tmpdir / 'test.dcd')
+        selection = u.trajectory[frames]
+        ref_positions = np.stack([ts.positions for ts in selection])
+        u.atoms.write(destination, frames=frames)
+
+        u_new = mda.Universe(destination)
+        new_positions = np.stack([ts.positions for ts in u_new.trajectory])
+
+        assert_array_almost_equal(new_positions, ref_positions)
+
+    @pytest.mark.parametrize('frames', (
+        [4],
+        [2, 3, 3, 1],
+        slice(2, 6, 1),
+    ))
+    def test_write_frame_iterator(self, u, tmpdir, frames):
+        destination = str(tmpdir / 'test.dcd')
+        selection = u.trajectory[frames]
+        ref_positions = np.stack([ts.positions for ts in selection])
+        u.atoms.write(destination, frames=selection)
+
+        u_new = mda.Universe(destination)
+        new_positions = np.stack([ts.positions for ts in u_new.trajectory])
+
+        assert_array_almost_equal(new_positions, ref_positions)
+
+    @pytest.mark.parametrize('extension', ('xtc', 'dcd', 'pdb', 'xyz'))
+    def test_write_frame_none(self, u, tmpdir, extension):
+        destination = str(tmpdir / 'test.' + extension)
+        u.atoms.write(destination, frames=None)
+        u_new = mda.Universe(destination)
+        new_positions = np.stack([ts.positions for ts in u_new.trajectory])
+        # Most format only save 3 decimals; XTC even has only 2.
+        assert_array_almost_equal(
+            u.atoms.positions[None, ...], new_positions, decimal=2
+        )
+
+    def test_write_frames_all(self, u, tmpdir):
```

Comment:
Add to the test something where you start from a frame in the middle of the trajectory and then check that after writing you are still at that frame.

Users will likely get confused if the act of writing changed "their own" trajectory – it's a bit confusing and a disadvantage of our architecture so we should make it as painless as possible.